homeassistant/components/prosegur/alarm_control_panel.py

```diff
@@ -0,0 +1,94 @@
+"""Support for Prosegur alarm control panels."""
+import logging
+
+from pyprosegur.auth import Auth
+from pyprosegur.installation import Installation, Status
+
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+)
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Prosegur alarm control panel platform."""
+    async_add_entities(
+        [ProsegurAlarm(entry.data["contract"], hass.data[DOMAIN][entry.entry_id])]
+    )
+
+
+class ProsegurAlarm(alarm.AlarmControlPanelEntity):
+    """Representation of a Prosegur alarm status."""
+
+    def __init__(self, contract: str, auth: Auth):
+        """Initialize the Prosegur alarm panel."""
+        self._state = None
+        self._changed_by = None
+
+        self._installation = None
+        self.contract = contract
+        self._auth = auth
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"contract {self.contract}"
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return the unique identifier of the alarm installation."""
+        return self.contract
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME
+
+    async def async_update(self):
+        """Update alarm status."""
+
+        try:
+            self._installation = await Installation.retrieve(self._auth)
+        except ConnectionError as err:
+            _LOGGER.error(err)
```

Comment:
Should the device become unavailable when that happens?

Reply:
Setting state to None now

Reply:
It's really unknown... it just means I can't connect to the cloud backend, not that the alarm is unavailable.

Reply:
Aah, the definition is the issue here.

The unknown is used as a state when the integration works, but are unfamiliar with the state it returns, the platform cannot provide a state at this moment (e.g., connected to a hub and device connected to that isn't responding); Or, in case of the assumed state, for example, we might not yet know a state.

The availability is used when the integration itself doesn't function. For example, in this case, the connection error.

Home Assistant will start behaving differently as well, it takes into account the integration cannot process or provide any state updates or changes at this time.

Reply:
I was actually assuming the cloud backend as a hub, so the device (alarm) doesn't respond / provide a state at this moment.

But that's OK, I can make it unavailable